asv_bench/benchmarks/algorithms.py

```diff
@@ -49,13 +58,21 @@ def setup(self, unique, sort, dtype):
             "datetime64[ns, tz]": pd.date_range(
                 "2011-01-01", freq="H", periods=N, tz="Asia/Tokyo"
             ),
+            "Int64": pd.array(np.arange(N), dtype="Int64"),
+            "boolean": pd.array(np.random.randint(0, 2, N), dtype="boolean"),
         }[dtype]
         if not unique:
             data = data.repeat(5)
         self.idx = data
+        if dtype in ("Int64", "boolean") and sort:
+            # sort is not a keyword on EAs
+            raise NotImplementedError
 
     def time_factorize(self, unique, sort, dtype):
-        self.idx.factorize(sort=sort)
+        if sort:
```

Comment:
isn't this redudant? since sort is a parameter?

`self.idx.factorize(sort=sort)`

Reply:
ExtensionArrays don't support the `sort` keyword, the other values are Index objects, which have that keyword. 
So the tests for `sort=True` are skipped above in case of `idx` being an EA

Reply:
this is very confusing then. I would separate the EAs out to a separate asv.

Reply:
> this is very confusing then. I would separate the EAs out to a separate asv.

Agree this is confusing. But I switched to use the factorize function in the hope to make this clearer, and to keep a single benchmark (the index method is just simply calling pd.factorize on itself, so this should benchmark the exact same thing). 
And that way, we can actually remove the skip for `sort` for EAs.

Reply:
@jreback updated. Much better now I think (and fine for a single benchmark class/function)